packages/telemed-ehr/app/src/hooks/useOttehrUser.tsx

```diff
@@ -142,14 +213,189 @@ export default function useOttehrUser(): OttehrUser | undefined {
         userInitials,
         lastLogin,
         profileResource: profile,
+        isERXPrescriber,
+        isPractitionerEnrolledInERX,
         hasRole: (role: RoleType[]) => {
           return userRoles.find((r) => role.includes(r.name as RoleType)) != undefined;
         },
       };
     }
     return undefined;
-  }, [lastLogin, profile, user, userInitials, userName]);
+  }, [lastLogin, isERXPrescriber, isPractitionerEnrolledInERX, profile, user, userInitials, userName]);
 }
 
 const MINUTE = 1000 * 60;
 const DAY = MINUTE * 60 * 24;
+
+interface StreetAddress {
+  street1: string;
+  street2?: string;
+  city: string;
+  state: string;
+  postal_code: string;
+}
+
+interface ERXEnrollmentProps {
+  providerId: Required<Practitioner['id']>;
+  address: StreetAddress;
+  npi?: string;
+  phone: string;
+  given_name: string;
+  family_name: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const useEnrollPractitionerInERX = () => {
+  const token = useAuthToken();
+  const profile = useOttehrUserStore((state) => state.profile);
+
+  return useMutation(
+    ['enroll-provider-erx'],
+    async (): Promise<void> => {
+      try {
+        const address: StreetAddress = {
```

Comment:
should this address be hardcoded?

Reply:
for now, yes, as it's required... (since it's for developers for now) but we would need to decide in next iteration where it should come from... same for the phone number.